packages/analytics/analytics-chart/src/components/TopNTable.vue

```diff
@@ -0,0 +1,268 @@
+          <div class="table-row">
+            <div class="column-1">
+              {{ i18n.t('topNTable.nameLabel') }}
+            </div>
+            <div class="column-2">
+              {{ columnName }}
+            </div>
+          </div>
+        </div>
+
+        <div
+          class="table-body"
+          data-testid="top-n-data"
+        >
+          <div
+            v-for="(entry, idx) in records"
+            :key="`entry-${idx}`"
+            class="table-row"
+            :data-testid="`row-${getId(entry)}`"
+          >
+            <div class="column-1">
+              <slot
+                name="name"
+                :record="{
+                  id: getId(entry),
+                  name: getName(entry),
+                  dimension: displayKey,
+                }"
+              >
+                {{ getName(entry) }}
+              </slot>
+            </div>
+            <div class="column-2">
+              &nbsp; {{ getValue(entry) }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </template>
+  </KCard>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import { AnalyticsExploreV3Result, AnalyticsExploreV3Meta, AnalyticsExploreRecord, RecordEvent } from '@kong-ui-public/analytics-utilities'
+// @ts-ignore - approximate-number no exported module
+import approxNum from 'approximate-number'
+import composables from '../composables'
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    default: '',
+  },
+  data: {
+    type: Object as PropType<AnalyticsExploreV3Result>,
+    required: true,
+  },
+  emptyStateTitle: {
+    type: String,
+    default: '',
+  },
+})
+
+const { i18n } = composables.useI18n()
+const meta = computed((): AnalyticsExploreV3Meta => props.data.meta)
+const records = computed((): AnalyticsExploreRecord[] => props.data.records)
+const hasData = computed((): boolean => !!(records.value?.length))
+const displayKey = computed((): string => {
+  if (!meta.value?.display) {
+    return ''
+  }
+
+  return Object.keys(meta.value.display)?.[0]
+})
+const displayRecord = computed(() => {
+  if (!displayKey.value) {
+    return {}
+  }
+
+  return meta.value.display[displayKey.value]
+})
+const columnKey = computed((): string => {
+  if (!meta.value?.metricNames?.length) {
+    return ''
+  }
+
+  return meta.value.metricNames[0]
+})
+const columnName = computed((): string => {
+  if (columnKey.value) {
+    return i18n.t(`chartLabels.${columnKey.value}` as any) || columnKey.value
+  }
+
+  return ''
+})
+
+const errorMessage = computed((): string => {
+  if (hasData.value) {
+    if (!meta.value) {
+      return i18n.t('topNTable.errors.meta')
+    }
+
+    if (!Object.keys(displayRecord.value).length) {
+      return i18n.t('topNTable.errors.display')
+    }
+
+    if (!columnKey.value) {
+      return i18n.t('topNTable.errors.metricNames')
+    }
+  }
+
+  return ''
```

Comment:
nit/refactor: same suggestion about exiting early with a string at the top to match the others